lib/nuget-parser/parsers/dotnet-core-v2-parser.ts

```diff
@@ -24,6 +19,42 @@ interface DotnetPackage {
 // Dependencies that starts with these are discarded
 const FILTERED_DEPENDENCY_PREFIX = ['runtime'];
 
+// The list of top level dependencies and transitive dependencies differ based on the target runtime we've defined.
```

Comment:
I know this looks crazy, but I don't have a better solution. Suggestions welcome.

It's pretty wild.

Take this [project.assets.json](https://github.com/snyk/snyk-nuget-plugin/blob/25ab60d6c9b0ee7cfb2a05dbe7bffc4b33fe93e4/test/fixtures/dotnetcore/netcoreapp20_target_monikers_differ/obj/project.assets.json) from the fixtures:

Targets:
https://github.com/snyk/snyk-nuget-plugin/blob/25ab60d6c9b0ee7cfb2a05dbe7bffc4b33fe93e4/test/fixtures/dotnetcore/netcoreapp20_target_monikers_differ/obj/project.assets.json#L3-L4

And frameworks (same file mind you!):
https://github.com/snyk/snyk-nuget-plugin/blob/25ab60d6c9b0ee7cfb2a05dbe7bffc4b33fe93e4/test/fixtures/dotnetcore/netcoreapp20_target_monikers_differ/obj/project.assets.json#L405-L406

Not sure if I am missing something obvious, but otherwise this was the best solution I could find.